lib/features/artifacts.js

```diff
@@ -128,7 +128,22 @@ export default class Artifacts {
       };
 
       try {
-        await uploadArtifact(taskHandler, stream, entryName, expiry, headers);
+        await new Promise((accept, reject) => {
+          let numRetries = taskHandler.runtime.retry.numRetries;
+          let retry = () => {
+            uploadArtifact(taskHandler, stream, entryName, expiry, headers)
+              .then(accept)
+              .catch((err) => {
+                if (numRetries--) {
+                  taskHandler.runtime.log('retrying artifact upload');
```

Comment:
We should include what the error was for good measure.  runtime.log takes an event and a object payload for things like error messages.